util/args_reader.go

```diff
@@ -0,0 +1,45 @@
+// Copyright (c) 2019, Google, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+//   Unless required by applicable law or agreed to in writing, software
+//   distributed under the License is distributed on an "AS IS" BASIS,
+//   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//   See the License for the specific language governing permissions and
+//   limitations under the License.
+
+package util
+
+import (
+	"bufio"
+	"errors"
+	"os"
+)
+
+func ReadArgsOrStdin(args []string) (string, error) {
+	if len(args) == 0 || args[0] == "" {
+		stdin := os.Stdin
+		fi, err := stdin.Stat()
+		if err != nil {
+			return "", err
+		}
+
+		pipedStdin := (fi.Mode() & os.ModeCharDevice) == 0
+		if fi.Size() <= 0 && !pipedStdin {
+			err = nil
+			return "", errors.New("no args or STDIN provided")
+		}
+
+		scanner := bufio.NewScanner(os.Stdin)
+		input := ""
+		for scanner.Scan() {
+			input = scanner.Text()
+		}
+		return input, nil
+	}
+	return args[0], nil
```

Comment:
Prefer to return early

Reply:
What does that mean in context?

Reply:
I mean try reducing nesting/indentation by reversing the conditional to look something like:

```
if len(args) != 0 {
  return args[0], nil
}

stdin := os.Stdin
fi, err :- stdin.Stat()
...

Reply:
Got it. Done.